test/controllers/products_controller_test.rb

```diff
@@ -170,35 +167,33 @@
   end
   
   describe "update" do 
-
     describe "while logged out" do 
       it "cannot update a product when logged out" do 
         # Arrange 
         @magician = products(:magician)
-
+        
         # Act
         get edit_product_path(@magician.id)
-
+        
         #Assert
         must_respond_with :redirect
         must_redirect_to root_path
       end
     end
-
-
+    
+    
     describe "while logged in" do 
       it "can update a product" do 
         # Arrange
         @magician = products(:magician)
         perform_login
         update_hash = {
-          product: {
-            name: "Houdini",
-            price: 150.99,
-            stock: 12,
-            retire: true 
-          }
-        }
+        product: {
+        name: "Houdini",
+        price: 150.99,
+        stock: 12,
+        retire: true 
+        }}
```

Comment:
Thanks for fixing some of the formatting mistakes in here but I think the current formatting is more readable than your updates for this part.